File: src/main/java/mods/eln/sim/electrical/heater/ElectricalResistorHeatThermalLoad.java
package mods.eln.sim.electrical.heater;

import mods.eln.sim.IProcess;
import mods.eln.sim.electrical.mna.component.Resistor;
import mods.eln.sim.thermal.ThermalLoad;

public class ElectricalResistorHeatThermalLoad implements IProcess {

    Resistor electricalResistor;
    ThermalLoad thermalLoad;

    public ElectricalResistorHeatThermalLoad(Resistor electricalResistor, ThermalLoad thermalLoad) {
        this.electricalResistor = electricalResistor;
        this.thermalLoad = thermalLoad;
    }

    @Override
    public void process(double time) {
        thermalLoad.PcTemp += electricalResistor.getP();
    }
}

File: src/main/java/mods/eln/sim/ElectricalResistorHeatThermalLoad.kt
package mods.eln.sim

import mods.eln.sim.mna.component.Resistor

class ElectricalResistorHeatThermalLoad(internal var electricalResistor: Resistor, internal var thermalLoad: ThermalLoad) : IProcess {

    override fun process(time: Double) {
        thermalLoad.PcTemp += electricalResistor.getPower()
    }
}
